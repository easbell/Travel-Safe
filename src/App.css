.App {
  text-align: center;
}

h1 {
  color: #000;
  opacity: 0.7;
  height: 100px;
  display: flex;
  justify-content: center;
  align-items: center;
  letter-spacing: 4.3px;
  margin: 25px auto 25px;
  font-size: 73px;
  font-weight: 400;
}

h1 span {
  font-weight: 500;
}

.saved {
  text-decoration: none;
  background-color: #000;
  opacity: .8;
  border-radius: 5px;
  padding: 13px;
  color: #fff;
  font-weight: 500;
  font-size: 21px;
  position: absolute;
  top: 25px;
  right: 35px;
  text-decoration: none;
}