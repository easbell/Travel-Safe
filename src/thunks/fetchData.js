import { setData, hasError } from '../actions';

export const fetchData = () => {
  return async (dispatch) => {
    try {
      const response = await fetch('http://localhost:3001/api/v1/data');
      if(!response.ok) {
        throw Error(response.statusText);
      }
      const data = await response.json();
      dispatch(setData(data[0]));
    } catch(error) {
      dispatch(hasError(error.message));
    }
  }
}