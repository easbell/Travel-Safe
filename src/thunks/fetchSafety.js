import { setSafety, hasError } from '../actions';
import { cleanSafety } from '../helpers/cleanSafety';

export const fetchSafety = () => {
  return async (dispatch) => {
    try {
      console.log('in safety')
      const response = await fetch('http://localhost:3001/api/v1/safety');
      if(!response.ok) {
        throw Error(response.statusText);
      }
      const data = await response.json();
      const cleanedData = cleanSafety(data);
      dispatch(setSafety(cleanedData));
    } catch(error) {
      dispatch(hasError(error.message));
    }
  }
}