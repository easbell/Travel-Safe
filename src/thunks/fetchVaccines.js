import { setVaccines, hasError } from '../actions';

export const fetchVaccines = () => {
  return async (dispatch) => {
    try {
      const response = await fetch('http://localhost:3001/api/v1/vaccines');
      if(!response.ok) {
        throw Error(response.statusText);
      }
      const data = await response.json();
      dispatch(setVaccines(data));
    } catch(error) {
      dispatch(hasError(error.message));
    }
  }
}