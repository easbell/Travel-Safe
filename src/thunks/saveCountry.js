import { setSaved, hasError } from '../actions';

export const saveCountry = (id) => {
  const option = {
    method: 'POST',
    body: JSON.stringify(id),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const url = 'http://localhost:3001/api/v1/data';
  console.log('in save, id:', JSON.stringify(id))
  return async (dispatch) => {
    try {
      const response = await fetch(url, option);
      if(!Response.ok) {
        throw Error(response.statusText);
      }
      dispatch(setSaved(response.body));
    } catch(error) {
      dispatch(hasError(error.message))
    }
  }
}