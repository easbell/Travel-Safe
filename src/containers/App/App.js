import {fetchData } from '../../thunks/fetchData';
import React, { Component } from 'react';
import CountriesContainer from '../../components/CountriesContainer/CountriesContainer';
import SavedCountries from '../SavedCountries/SavedCountries';
import { connect } from 'react-redux';
import { Route, NavLink } from 'react-router-dom';
import CountryDetails from '../CountryDetails/CountryDetails';
import './App.css';

export class App extends Component {
  componentDidMount() {
    this.props.fetchData();
  }

  render() {
    const { data } = this.props;
    return (
      <div className="App">
        <h1>Travel<span className='bold'>Safe</span></h1>
        <Route exact path='/' render={() => (
          <div>
            <NavLink to='/saved'className='saved'>Saved</NavLink>
            <CountriesContainer countries={data}/>
          </div>
        )} />
        <Route exact path='/saved' render={() => (
            <div>
              <NavLink to='/'className='saved'>Home</NavLink>
              <SavedCountries />
            </div>
        )} />
        <Route path='/details/:id' render={({ match }) => {
          const { id } = match.params;
          const selectedCountry = data.find(country => {
            return country.id === id
          })
          if(selectedCountry) {
            return <CountryDetails {...selectedCountry} />
          }
        }} />
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  fetchData: () => dispatch(fetchData())
});

export const mapStateToProps = (state) => ({
  data: state.data
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
