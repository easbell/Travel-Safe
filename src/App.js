import {fetchSafety } from './thunks/fetchSafety';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import './App.css';

class App extends Component {
  componentDidMount() {
    this.props.fetchSafety();
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1>Travel Safe</h1>
        </header>
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  fetchSafety: () => dispatch(fetchSafety())
});

export default connect(null, mapDispatchToProps)(App);
