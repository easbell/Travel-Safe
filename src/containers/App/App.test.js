import React from 'react';
import { App, mapDispatchToProps, mapStateToProps } from './App';
import { shallow } from 'enzyme';
import { fetchData } from '../../thunks/fetchData';

jest.mock('./thunks/fetchData')

describe('App', () => {
  let wrapper;
  let mockFn;
  beforeEach(() => {
    mockFn = jest.fn();
    wrapper = shallow(<App fetchData={mockFn}/>)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call fetchData when mounted', () => {
    wrapper.instance().componentDidMount();
    expect(mockFn).toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with fetchData action', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = fetchData();
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.fetchData();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

  describe('mapStateToProps', () => {
    it('returns correct props', () => {
      const mockState = {
        data: ['some mock data'],
        other: 'other'
      }
      const expectedProps = {
        data: ['some mock data']
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expectedProps)
    });
  });
});