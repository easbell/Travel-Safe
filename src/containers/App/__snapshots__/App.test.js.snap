// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    fetchData={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          className="header"
          to="/"
        >
          <h1>
            Travel
            <span
              className="bold"
            >
              Safe
            </span>
          </h1>
        </NavLink>,
        <Route
          exact={true}
          path="/"
          render={[Function]}
        />,
        <Route
          exact={true}
          path="/saved"
          render={[Function]}
        />,
        <Route
          path="/details/:id"
          render={[Function]}
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <h1>
            Travel
            <span
              className="bold"
            >
              Safe
            </span>
          </h1>,
          "className": "header",
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Travel",
              <span
                className="bold"
              >
                Safe
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Travel",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Safe",
                "className": "bold",
              },
              "ref": null,
              "rendered": "Safe",
              "type": "span",
            },
          ],
          "type": "h1",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/saved",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/details/:id",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            className="header"
            to="/"
          >
            <h1>
              Travel
              <span
                className="bold"
              >
                Safe
              </span>
            </h1>
          </NavLink>,
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/saved"
            render={[Function]}
          />,
          <Route
            path="/details/:id"
            render={[Function]}
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <h1>
              Travel
              <span
                className="bold"
              >
                Safe
              </span>
            </h1>,
            "className": "header",
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Travel",
                <span
                  className="bold"
                >
                  Safe
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Travel",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Safe",
                  "className": "bold",
                },
                "ref": null,
                "rendered": "Safe",
                "type": "span",
              },
            ],
            "type": "h1",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/saved",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/details/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
